### PR TITLE
Print the Hex string of the 16 first bytes of blob when ACHILLES_DML_STATEMENT is enabled

### PR DESCRIPTION
**Pull request by bric3**
This PR allows to print the hexadecimal string of `byte` array or a `ByteBuffer` parameter when `ACHILLES_DML_STATEMENT` logger is enabled.

This is especially useful when debugging to find consistency issues, where the blob contains valuable information.

The impact should be fairly limited for two reasons : 

- `ACHILLES_DML_STATEMENT` is not to be enabled on prod
- The proposed change only _peek_ at the first 16 first bytes (like 2 longs), at the moment this limit is hardcoded, but it could improved later.
  If longer than 16 bytes 3 dots are appended signaling the reader that it was a longer blob in the query.

### CODE REVIEW

File: achilles-core/src/main/java/info/archinnov/achilles/internal/helper/LoggerHelper.java [hunk 1]
```diff
@@ -31,4 +38,22 @@ public String apply(Field field) {
...
+        for (int valuePos = 0; valuePos < boundValues.size(); valuePos++) {
+            Object boundValue = boundValues.get(valuePos);
+            if (boundValue instanceof ByteBuffer) {
+                ByteBuffer byteBuffer = (ByteBuffer) boundValue;
+                byte[] extracted = new byte[min(byteBuffer.remaining(), HEX_STRING_LOG_LIMIT)];
+                byteBuffer.get(extracted).rewind();
+                boundValues.set(valuePos, Bytes.toHexString(extracted) + ((byteBuffer.remaining() > HEX_STRING_LOG_LIMIT) ? "..." : ""));
+            } else if (boundValue instanceof byte[]) {
+                byte[] byteArray = (byte[]) boundValue;
+                boundValues.set(valuePos, Bytes.toHexString(((byteArray.length > HEX_STRING_LOG_LIMIT) ? Arrays.copyOfRange(byteArray, 0, HEX_STRING_LOG_LIMIT) : byteArray))  + ((byteArray.length > HEX_STRING_LOG_LIMIT) ? "..." : ""));
```
Comment (obourgain) [inline achilles-core/src/main/java/info/archinnov/achilles/internal/helper/LoggerHelper.java]: can't you put the ```? "..." : ""``` logic that appearrs in both branches and even worse in the else as a nested ternary op in a method ?

File: achilles-core/src/test/java/info/archinnov/achilles/LoggerTestUtils.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import static org.mockito.Mockito.mock;
+import java.lang.reflect.Field;
+import org.apache.log4j.Appender;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.impl.Log4jLoggerAdapter;
+
+public abstract class LoggerTestUtils {
+    public static Appender interceptDMLStatementViaMockedAppender() {
```
Comment (obourgain) [inline achilles-core/src/test/java/info/archinnov/achilles/LoggerTestUtils.java]: You are modifiying a shared variable, you should restore its state after the test, maybe with a junit rule

**Comment by obourgain** [comment]
Maybe also output the length of the byte buffer ?

**Comment by bric3** [comment]
This PR also has the benefit of aligning the behavior of CQLSH and Achilles, for example 

```
cqlsh:my_ks> select * from dumb_table ;

  key | value
-----+----------------------------------------------------------------------------
   1 | 0x6a686b646e666b6c73676e6c7372
```

While DML statements with are logged this way : 

```
2015-03-19 07:07:21,206 [http-bio-8080-exec-5] DEBUG ACHILLES_DML_STATEMENT - Bound statement : [INSERT INTO contacts.contacts_by_id(id,value) VALUES (:id,:value) USING TTL :ttl;] with CONSISTENCY LEVEL [LOCAL_QUORUM]
2015-03-19 07:07:21,206 [http-bio-8080-exec-5] DEBUG ACHILLES_DML_STATEMENT -    bound values : [1985101, java.nio.HeapByteBuffer[pos=0 lim=4 cap=4]]
```


With this change we will see : 

```
2015-03-19 07:07:21,206 [http-bio-8080-exec-5] DEBUG ACHILLES_DML_STATEMENT - Bound statement : [INSERT INTO contacts.contacts_by_id(id,value) VALUES (:id,:value) USING TTL :ttl;] with CONSISTENCY LEVEL [LOCAL_QUORUM]
2015-03-19 07:07:21,206 [http-bio-8080-exec-5] DEBUG ACHILLES_DML_STATEMENT -    bound values : [1985101, 0x6a686b646e666b6c...]
```

File: achilles-core/src/test/java/info/archinnov/achilles/LoggerTestUtils.java (hunk 2 above)
Comment (bric3) [inline achilles-core/src/test/java/info/archinnov/achilles/LoggerTestUtils.java]: Good idea, thanks

**Pull request merged**